src/check_api_reachability.py

```
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+#
+# Nagios API reachability check
+#
+# This is a Nagios check to check our external services,
+# that have to be reachable by our game servers.
+# Checks by default for HTTP and HTTPS connections.
+# Accepted HTTP codes: 200, 201, 301, 302, 307, 403
+#
+# Returns
+#
+# * Critical when at least one URL is not reachable
+#
+# Copyright (c) 2019 InnoGames GmbH
+#
+
+from argparse import ArgumentParser
+# from requests import (
+#     get,
+#     RequestException,
+# )
+
+from urllib.request import urlopen
+from urllib.error import HTTPError
+
+
+class ExitCodes:
+    ok = 0
+    critical = 2
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument(
+        '--url', dest='url_list',
+        type=str, action='append', required=True
+        )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    unreachable_urls = []
+    failure = False
+    failure_count = 0
+
+    for url in args.url_list:
+        check, domain = check_url(url)
+        if not check:
+            unreachable_urls.append(domain)
+            failure = True
+            failure_count += 1
+
+    if not failure:
+        output = 'OK: All URLs are reachable.'
+        exit_code = ExitCodes.ok
+    else:
+        output_list = ([domain['url'] + ' - ' + str(domain['error'])
+                        for domain in unreachable_urls])
+        output = 'CRITICAL: {} URLs not reachable. {}'.format(
+                failure_count, '; '.join(output_list)
+                )
+        exit_code = ExitCodes.critical
+
+    print(output)
+    exit(exit_code)
+
+
+def check_url(url, accepted_codes={200, 201, 301, 302, 307, 403}):
+    try:
+        request = urlopen(url)
+    except HTTPError:
+        return False, {'url': url, 'error': 'exception'}
```

why would you return the url here? the url is already known because it was given as an argument

---

also you can just return the error if there is one, otherwise return None. no need for an additional boolean

---

The script takes multiple URLs. It's there so we can identify which one is the troublesome.

---

this function takes only one url.